### DISCUSSION TITLE
Why is Bromite falling behind in updates

### DISCUSSION THREAD

**Discussion by ghost** [bromite/bromite#1745] [Q&A] [answered]
Why is Bromite falling behind in updates, Chromium 97 has released most chromium forks have updated to chromium 97, Bromite is still on chromium 96.

**Comment by syphyr** [comment]
The current version of Bromite is vulnerable to a RCE

**Reply by csagan5** [reply]
You must be referring to this one: https://chromereleases.googleblog.com/2022/01/stable-channel-update-for-desktop_19.html

**Reply by syphyr** [reply]
I am referring the to second RCE vulnerability found in chromium's v8 dependency involving libpac.

**Reply by syphyr** [reply]
Fixing this issue does not require a major revision update. Maybe in the meantime, updating to latest 96.0.4664.xxx would be sufficient.

**Reply by csagan5** [reply]
@syphyr please provide a CVE?

**Reply by syphyr** [reply]
CVE-2021-4102

**Reply by SkewedZeppelin** [reply]
I counted it at 86 known vulnerabilities.

For reference:
https://divestos.org/misc/ch-dates.txt
https://divestos.org/misc/ffa-dates.txt

Edit: 98 is now out
new count: 20 + 5 + 35 + 26 + 27 = 113

**Reply by ghost** [reply]
> second RCE vulnerability found in chromium's v8 dependency

does disabling javascript solve this?

**Reply by syphyr** [reply]
It would require disabling proxy settings and using a pac proxy file.

**Reply by ghost** [reply]
> It would require disabling proxy settings and using a pac proxy file.

i don't use a proxy, so how to disable proxy settings?

**Reply by syphyr** [reply]
Then proxy is disabled. You can probably ignore the RCE I mentioned, although, there seem to be more cve than this.

**Reply by csagan5** [reply]
> CVE-2021-4102

I could not figure out back then if this applies to all users or only those with JIT enabled, but thanks for providing the CVE reference.

> I counted it at 86 known vulnerabilities.
> 
> For reference: https://divestos.org/misc/ch-dates.txt https://divestos.org/misc/ffa-dates.txt
> 
> Edit: 98 is now out new count: 20 + 5 + 35 + 26 + 27 = 113

This is a superficial way to talk about security. Not all of these affect Chromium browsers on Android and not all of those affect Bromite either.

I applaud the effort done by the DivestOS people about tracking the security patches but wish that upstream (Chromium) would mark them better and also clearly tell which platforms and components are affected.

**Comment by csagan5** [comment, marked as answer]
That is easy to answer: it is falling behind in updates because there is not enough developers contributing and it requires quite a bit of time from the current ones.

If you are a developer you can contribute with bugfixes and feature development; as an user you can help with testing and/or donations.

Unfortunately, complaining alone will not help the project much :disappointed: if complaints could be used to pay server bills and fix bugs it would be a much better project by now!

**Reply by ghost** [reply]
Thanks for your kind reply.

**Reply by SkewedZeppelin** [reply]
Perhaps instead of 8 of us maintaining each our own builds, we should all work together on a common patchset repository where we can all maintain them.

**Reply by binarious** [reply]
IMHO there should be a red banner on the GitHub readme and on the website stating that you don't have enough contributors to include recent security fixes. For one that would make people aware of the fact that you need maintainers and it communicates that the browser could be vulnerable. I thought Bromite might not be affected by the recent RCEs, because it's silent in this repository about it.

**Reply by csagan5** [reply]
@vortaj it is a product when you pay for it; this is not a product.

@binarious I do not need maintainers, I need contributors; I will think about a better way to expose which CVEs affect Bromite in future. Thanks for your opinions.

**Reply by chrislogan-foss** [reply]
> Perhaps instead of 8 of us maintaining each our own builds, we should all work together on a common patchset repository where we can all maintain them.

Then do so

**Reply by csagan5** [reply]
> Perhaps instead of 8 of us maintaining each our own builds, we should all work together on a common patchset repository where we can all maintain them.

We discussed this some time ago: https://github.com/Eloston/ungoogled-chromium/discussions/1395

**Comment by SkewedZeppelin** [comment]
@csagan5 
If you aren't aware /e/OS ships a [rebranded](https://gitlab.e.foundation/e/apps/browser) version of Bromite.
Perhaps you can reach out to them for them to sponsor you or push them to contribute? Assuming you haven't spoke with them already.
Just throwing it out there.

I haven't reviewed it too much, but it seems @uazo has been repeatedly doing the rebase work as well.

**Reply by csagan5** [reply]
I am aware since when they started; I do not consider rebranding an open source project as a friendly move. They have always been free to contribute but never contributed back.
